scripts/west_commands/runners/stm32cubeprogrammer.py

```diff
@@ -273,8 +273,12 @@ def flash(self, **kwargs) -> None:
         if self._use_elf:
             # Use elf file if instructed to do so.
             dl_file = self.cfg.elf_file
-        elif self.cfg.bin_file is not None and self._download_address is not None:
-            # Use bin file if a binary is available and --download-address provided
+        elif (self.cfg.bin_file is not None and
+                (self._download_address is not None or
+                 ("usb" in str(self._port) and self._download_modifiers is not None))):
```

Comment:
hey this looks a bit suspicious

```py
        parser.add_argument(
            "--port",
            type=str,
            required=True,
            help="Interface identifier, e.g. swd, jtag, /dev/ttyS0...",
        )
```

is this going to work? I'd expect it to get stuff like `ttyUSB0` or `ttyACM0` or whatever alias a user may set up. I imagine you tested it with some `/dev/serial/by-<something>`

Maybe you can try to guess the list of ports using pyserial instead?

Anyway let's hold this for 4.2 + backport

Reply:
Agree this is not ideal, this is why I added `self._download_modifiers is not None` which is STM32N6 specific.
See: https://github.com/zephyrproject-rtos/zephyr/blob/f3630157e03fa322787f264b1047f91da5c2af8c/boards/st/stm32n6570_dk/board.cmake#L5-L6

Reply:
ok so it's not a filesystem path? but it could be a filesystem path on other platforms? it's all very confusing, maybe https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/west_commands/runners/stm32cubeprogrammer.py#L157 could use some tweak too

Reply:
We're trying to match STM32CubeProgrammer API.
See https://www.st.com/resource/en/user_manual/um2237-stm32cubeprogrammer-software-description-stmicroelectronics.pdf
> $ 3.2.1 Connect command:
port=\<Portname Interface identifier\>, ex COMx (for Windows), /dev/ttySx for Linux), usbx for USB interface, SPI, 
I2C and CAN for, respectively, SPI, I2C and CAN interfaces

Reply:
also how much you want this for 4.1? would you consider this a regression or is it something that never really worked?

Reply:
The initial feature was introduced in https://github.com/zephyrproject-rtos/zephyr/pull/84869 and broke with https://github.com/zephyrproject-rtos/zephyr/pull/85332.
The feature is required for test automation and useful for development phase (allows to flash w/o touching jumpers) so that would be ideal to have, but I would understand this is a bit late.

Reply:
regression during stabilization sounds like release blocker to me